docs/reference/search/search-your-data/knn-search.asciidoc

```diff
@@ -1068,100 +1068,73 @@ NOTE: Approximate kNN search always uses the
+It will then use the original vectors to rescore them, and return the top `k` results.
+
+Here is an example of using the `rescore_vector` option with the `num_candidates_factor` parameter:
 
-. Use the <<search-search,search API>> to run a `script_score` query containing
-a <<vector-functions,vector function>>.
-+
-TIP: To limit the number of matched documents passed to the vector function, we
-recommend you specify a filter query in the `script_score.query` parameter. If
-needed, you can use a <<query-dsl-match-all-query,`match_all` query>> in this
-parameter to match all documents. However, matching all documents can
-significantly increase search latency.
-+
 [source,console]
 ----
-POST product-index/_search
+POST image-index/_search
 {
-  "query": {
-    "script_score": {
-      "query" : {
-        "bool" : {
-          "filter" : {
-            "range" : {
-              "price" : {
-                "gte": 1000
-              }
-            }
-          }
-        }
-      },
-      "script": {
-        "source": "cosineSimilarity(params.queryVector, 'product-vector') + 1.0",
-        "params": {
-          "queryVector": [-0.5, 90.0, -10, 14.8, -156.0]
-        }
-      }
+  "knn": {
+    "field": "image-vector",
+    "query_vector": [-5, 9, -12],
+    "k": 10,
+    "num_candidates": 100,
+    "rescore_vector": {
+      "num_candidates_factor": 2.0
     }
-  }
+  },
+  "fields": [ "title", "file-type" ]
 }
 ----
 //TEST[continued]
+// TEST[s/"k": 10/"k": 3/]
+// TEST[s/"num_candidates": 100/"num_candidates": 3/]
+
+This example will:
+
+* Search using approximate kNN with `num_candidates` set to 200 (`num_candidates` * `num_candidates_factor`).
+* Rescore the top 200 candidates per shard using the original, non quantized vectors.
+* Merge the rescored canddidates from all shards, and return the top 10 (`k`) results.
 
 [discrete]
-[[dense-vector-knn-search-reranking]]
-==== Oversampling and rescoring for quantized vectors
+[[dense-vector-knn-search-reranking-rescore-additional]]
+===== Additional rescoring techniques
 
-All forms of quantization will result in some accuracy loss and as the quantization level increases the accuracy loss will also increase.
-Generally, we have found that:
-- `int8` requires minimal if any rescoring
-- `int4` requires some rescoring for higher accuracy and larger recall scenarios. Generally, oversampling by 1.5x-2x recovers most of the accuracy loss.
-- `bbq` requires rescoring except on exceptionally large indices or models specifically designed for quantization. We have found that between 3x-5x oversampling is generally sufficient. But for fewer dimensions or vectors that do not quantize well, higher oversampling may be required.
+The following sections provide additional ways of rescoring:
+
+[discrete]
+[[dense-vector-knn-search-reranking-rescore-section]]
+====== Use the `rescore_vector` section for top-level kNN search
```

Comment:
Should this be `rescore` instead of `rescore_vector`?

Reply:
Thanks for the catch - fixed!